chore(deps): bump spring-core and spring-test to 6.0.5

Bronstrom opened the pull request:
This handles updating both spring-core and
spring-test in one PR. As far as it seems
spring-core and spring-test have consistently
follow the same version in maven, so
spring-test has been made to reference
spring-core in attempt to avoid
multiple merge requests.

Bronstrom committed:
fix: Bump spring-core and spring-test to 6.0.5

This handles updating both spring-core and
spring-test in one PR. As far as it seems
spring-core and spring-test have consistently
follow the same version in maven, so
spring-test has been made to reference
spring-core in attempt to limit avoid
multiple merge requests.

laurelmay commented on pom.xml:
```diff
@@ -19,8 +19,8 @@
 		<spring-boot-starter-data-jpa.version>3.0.2</spring-boot-starter-data-jpa.version>
-		<spring-core.version>6.0.4</spring-core.version>
-		<spring-test.version>6.0.4</spring-test.version>
+		<spring-core.version>6.0.5</spring-core.version>
+		<spring-test.version>${spring-core.version}</spring-test.version>
```
If we we're doing this, can we just have a:

```xml
<spring.version>6.0.5</spring-version>
```

And then we can just replace all occurrences of `spring-core` and `spring-test` with this?

Can we also see if there's any reason _not_ to do this? (How often do `spring-core` and `spring-test` have different version numbers?)

Bronstrom replied:
I've gone ahead and replaced all occurrences of `spring-core.version` and `spring-test.version` with `spring.version`. I don't see any reason not to do this, as `spring-core` and `spring-test` are always in line with each other. I've left the `scope`s and `artifactID`s to include `core` and `test`. Let me know if it makes sense to just align these to `spring` as well.

Bronstrom committed:
fix: swap spring-core & spring-test for spring

Due to spring-core and spring-test having aligned
versions, these dependencies have been swapped
for spring instead.
